src/compiler/checker.ts

```diff
@@ -33109,27 +33111,33 @@ namespace ts {
                 return;
             }
 
-            const testedFunctionSymbol = getSymbolAtLocation(testedNode);
-            if (!testedFunctionSymbol) {
+            const testedSymbol = getSymbolAtLocation(testedNode);

```

Comment:
`getSymbolAtLocation` is intended for interactive use by the services layer. It usually boils down to `resolveEntityName` or even `resolveName`, which is what you should use inside the checker.

Reply:
It was used in the previous version.

Reply:
Is it required to use some additional helpers to resolve names for cases like `this.required` || `x.foo.bar`?

Reply:
After our discussion, I think it's OK to leave this; there aren't any equivalent helpers already, and it's used in quite a few places throughout the checker already.

It might be worth a cleanup pass later, to holistically find a replacement, but that would be a lot of work.